services/motion/builtin/builtin.go

```diff
@@ -231,24 +236,109 @@ func (ms *builtIn) MoveOnGlobe(
 	kinematicsOptions.GoalRadiusMM = math.Min(motionCfg.PlanDeviationM*1000, 3000)
 	kinematicsOptions.HeadingThresholdDegrees = 8
 
-	plan, kb, err := ms.planMoveOnGlobe(
-		ctx,
-		componentName,
-		destination,
-		movementSensorName,
-		obstacles,
-		kinematicsOptions,
-		extra,
-	)
-	if err != nil {
-		return false, fmt.Errorf("error making plan for MoveOnMap: %w", err)
-	}
+	movementSensor, ok := ms.movementSensors[movementSensorName]
+	if !ok {
+		return false, resource.DependencyNotFoundError(movementSensorName)
+	}
+
+	// TODO: if we disentangle creation of the kinematic base from the plan call we make we don't need this and can do this once
+	var kinematicBase kinematicbase.KinematicBase = nil
+
+	// Shared variables for planning and execution threads, are all guarded with a mutex
+	var planMu sync.Mutex
+	var currentPlan [][]referenceframe.Input = nil
+
+	// Shared variable between threads that can trigger replanning and planning thread
+	var replanFlag atomic.Bool
+	replanFlag.Store(true)
+
+	var successFlag atomic.Bool
+	successFlag.Store(false)
+	errChan := make(chan error)
+
+	cancelCtx, cancelFn := context.WithCancel(context.Background())
+	ms.cancelFn = cancelFn
+
+	ms.backgroundWorkers.Add(1)
+	goutils.ManagedGo(func() {
+
+		positionPollingTicker := time.NewTicker(time.Duration(1000/motionCfg.PositionPollingFreqHz) * time.Millisecond)
+		obstaclePollingTicker := time.NewTicker(time.Duration(1000/motionCfg.ObstaclePollingFreqHz) * time.Millisecond)
+
+		for !successFlag.Load() && ctx.Err() == nil {
+			select {
+			case <-positionPollingTicker.C:
+				// TODO: theres nothing stopping us from making another thread before the previous has completed
+				ms.backgroundWorkers.Add(1)
+				goutils.ManagedGo(func() {
+					fmt.Println("position poll")
+					current, _, err := movementSensor.Position(cancelCtx, nil)
+					if err != nil {
+						errChan <- err
+						return
+					}
+					if spatialmath.GeoPointToPose(current, destination).Point().Norm() <= kinematicsOptions.PlanDeviationThresholdMM {
+						successFlag.Store(true)
+					}
+
+					// TODO: the function that actually monitors position - and pass context in
+				}, ms.backgroundWorkers.Done)
+			case <-obstaclePollingTicker.C:
+				// TODO: theres nothing stopping us from making another thread before the previous has completed
+				ms.backgroundWorkers.Add(1)
+				goutils.ManagedGo(func() {
+					fmt.Println("obstacle poll")
+					replanFlag.Store(true)
+
+					// this is gonna get removed when I pass context along the way
+					if !utils.SelectContextOrWait(cancelCtx, 1*time.Millisecond) {
+						fmt.Println("I've been cancelled")
+					}
+
+					// TODO: the function that actually monitors obstacles - and pass context in
+				}, ms.backgroundWorkers.Done)
+			default:
```

Comment:
[q]: does `default:` need to be here?

Reply:
yes, otherwise it will block on one of the tickers firing, which is not the condition we want to move on with.  We want to be polling the replanning flag which may or may not be set by the threads but either way it will not happen instantaneously